tests/test_2198_almost_equal.py

```diff
@@ -147,5 +147,5 @@ def test_numpy_array():
 
 def test_typetracer():
     array = ak.Array([[[1, 2, 3]], [[5, 4]]], backend="typetracer")
-    with pytest.raises(TypeError):
+    with pytest.raises(NotImplementedError):
```

Comment:
This should already have been failing, but `unknown_length == X` checks were raising `TypeError`s.